src/fmeval/eval_algorithms/common.py

```diff
@@ -98,10 +99,12 @@ def evaluate_dataset(
         Currently, only MEAN is supported.
     :param save: If set to true, prompt responses and scores will be saved to a file.
         The path that this file is stored at is configured by `eval_results_path`.
-
+    :param validate_columns: A boolean indicating if validation should be done on the dataset columns.
+        The model_input column is validated if a model is provided, and the model_output column is
+        validated if no model is provided.
     :return: An EvalOutput object encapsulating the results of the evaluation.
     """
-    if eval_name != EvalAlgorithm.CONTEXT_QUALITY.value:
+    if validate_columns:
```

Comment:
I don't think we want to have this outermost `if` statement. What I proposed in a previous comment was to add parameters representing 
1. The list of columns to validate in the `if model` case (currently, we always call `validate_dataset(dataset, [DatasetColumns.MODEL_INPUT.value.name])`, but this could potentially change (unlikely) depending on the eval algo).
2. The list of columns to validate in the `else` case (currently, we always call `validate_dataset(dataset, [DatasetColumns.MODEL_OUTPUTvalue.name])`), but for your new eval algo, you'd want to pass an empty list, since your algo actually doesn't need model outputs in the initial dataset. Note that you might need to update `validate_dataset` to handle the empty list case.

The current `validate_columns` boolean is a bit incorrect, as the code under `if validate_columns` does more than just validate columns. If you want to continue using this boolean, it would be better to add the `if validate_columns` right before the `try` statements instead.

Reply:
Discussed offline, will add a `col_to_validate` parameter which defaults to `DatasetColumns.MODEL_OUTPUT.value.name`, and set it to `None` for this usecase.